contracts/PvpArena.sol

```diff
@@ -35,54 +41,119 @@ contract PvpArena is Initializable, AccessControlUpgradeable {
     int128 public extraCostPerLevel;
 
     /// @dev fighters available by tier (1-10, 11-20, etc...)
-    mapping (uint8 => Fighter[]) private fightersByTier;
+    mapping(uint8 => Fighter[]) private fightersByTier;
     /// @dev fighters by player address
-    mapping (address => Fighter[]) private fightersByPlayer;
+    mapping(address => Fighter[]) private fightersByPlayer;
+
+    ///@dev mapping to track the characters used
+    mapping(uint256 => bool) public isCharacterUsed;
+
+    ///@dev mapping to track the weapons used
+    mapping(uint256 => bool) public isWeaponUsed;
+
+    ///@dev mapping to track the shields used
+    mapping(uint256 => bool) public isShieldUsed;
+
+    modifier beforeEnteringArena(
```

Comment:
I have a question here, this seems to me like we'll never reuse this particular check ever, because it is explicitly coupled with the `enterArena` function.

I'd rather divide these into reusable checks or simply add this logic into the function itself.